horace_core/sqw/PixelData/@PixelData/private/binary_op_double_.m

```diff
@@ -0,0 +1,110 @@
+function obj = binary_op_double_(obj, double_array, binary_op, flip, npix)
+%% BINARY_OP_DOUBLE_ perform a binary operation between this PixelData object
+% and an array
+%
+validate_input_array(obj, double_array, npix);
+
+if isempty(npix)
```

Comment:
there is a lot of code in the `if` and `else` blocks here; helper functions may make things clearer